processor/t60_processor.py

```diff
@@ -26,75 +35,231 @@ def extract_s3_info(object_url: str) -> [str, str]:
         match = re.match(regex, object_url)
         bucket, object_key = match.group(1), unquote(match.group(2))
         return [bucket, object_key]
+    
+    @staticmethod
+    def get_object_url(bucket: str, object_key: str) -> str:
+        """
+            Returns a valid S3 Object URL
+        """
+        return f'https://s3.amazonaws.com/{bucket}/{object_key}'
+
+    @staticmethod  
+    def extract_parent_key(object_key: str) -> str:
+        """
+        Returns the parent object key
+        """
+        return '/'.join(object_key.split('/')[:-1])
 
 
 class Processor:
 
     def __init__(self) -> None:
         self._S3_READ_CLIENT = client(
             "s3",
-            region_name=environ.get('AWS_BUCKET_REGION')
+            region_name=environ.get('AWS_BUCKET_REGION'),
+            aws_access_key_id=environ.get('AWS_ACCESS_KEY_ID'),
+            aws_secret_access_key=environ.get('AWS_SECRET_ACCESS_KEY')
         )
+        # getting the environment variables
         self.skip_frame = environ.get('SKIP_FRAME', 3)
         self.video_path = environ.get('VIDEO_PATH')
         self.camera_path = environ.get('CAMERA_PATH')
-        self.output_path = ''
+        self.date = environ.get('DATE')
+        self.EBS_PATH = environ.get('EBS_PATH', None)
+        self.batch_size = environ.get('BATCH_SIZE', 10)
+        self.num_threads = environ.get('NUM_THREADS', 2)
+        
+        self.output_path = 'messagepack.msg'
+        self.bucket = ''
+        self.parent_key = ''
+        self.dest_path = './360Images/'
+        self.coordinates_file = 'coordinates.json'
```

Comment:
```suggestion
        self.coordinates_file = 'Coordinates/coordinates.json'
```

Because within the coordinates file you are keeping the relative path as '../360Images/key' This is assuming that coordinates.json file itself is within a folder

Reply:
![image](https://github.com/powernai/stella_vslam/assets/70714558/1f2a3041-5011-4c04-8026-466b338ef156)

currently this is the structure and it's getting copied correctly
i am handling upload of coordinates.json file seperately
 #upload coordinates.json file in s3
        file_obj_path = os.path.basename(self.coordinates_file)
        with open(file_obj_path, 'rb') as fp:
            self._S3_READ_CLIENT.upload_fileobj(Fileobj=fp, Bucket=self.bucket, Key=self.parent_key + f"/{self.date}/{self.coordinates_file}")

Reply:
- The first one should not be there I guess, was that a mistake or will it always create that?
- Also, the location of the date folder should be created in the same path as the video file that should be the parent_key I guess
- As long as the coordinate file reflect the image relative path correctly we are good, if we are going with this we need small change in the front end. Currently Potree is adding coordination.json to the provided path itself this needs change you can change that when you work on [#362](https://github.com/powernai/cpms/issues/362)

Reply:
resolved the comments